chain/cosmos/ics.go

```diff
@@ -0,0 +1,392 @@
+package cosmos
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"testing"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/types"
+	govv1beta1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types" // nolint:staticcheck
+	ccvclient "github.com/cosmos/interchain-security/v5/x/ccv/provider/client"
+	"github.com/icza/dyno"
+	"github.com/strangelove-ventures/interchaintest/v8/dockerutil"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"golang.org/x/mod/semver"
+	"golang.org/x/sync/errgroup"
+
+	stakingttypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+const (
+	icsVer330 = "v3.3.0"
+	icsVer400 = "v4.0.0"
+)
+
+// FinishICSProviderSetup sets up the base of an ICS connection with respect to the relayer, provider actions, and flushing of packets.
+// 1. Stop the relayer, then start it back up. This completes the ICS20-1 transfer channel setup.
+//   - You must set look-back block history >100 blocks in [interchaintest.NewBuiltinRelayerFactory].
+//
+// 2. Get the first provider validator, and delegate 1,000,000denom to it. This triggers a CometBFT power increase of 1.
+// 3. Flush the pending ICS packets to the consumer chain.
+func (c *CosmosChain) FinishICSProviderSetup(t *testing.T, ctx context.Context, r ibc.Relayer, eRep *testreporter.RelayerExecReporter, ibcPath string) {
+	// Restart the relayer to finish IBC transfer connection w/ ics20-1 link
+	require.NoError(t, r.StopRelayer(ctx, eRep))
+	require.NoError(t, r.StartRelayer(ctx, eRep, ibcPath))
```

Comment:
I tried with just StartRelayer but it does not work, likely since the relayer is already running in the background. I don't feel a RestartRelayer is needed on the relayer interface for this.